src/structs.jl

```diff
@@ -77,7 +77,7 @@ end
 """
 Function application ("call by name", no tricks allowed) in mathematical expression
 """
-struct MathApply <: Math
+mutable struct MathApply <: Math
```

Comment:
please do not use mutable structures. The Model represents something that's in a file, which should not be mutable.

Reply:
Well, I need to change `Math` (and thus `Reaction`s) in `to_extensive_math!()`, and `Species` in `to_initial_amounts()`. It seems a bit cumbersome to write all this without mutating the `Math`, `Reaction`s and `Species`, but I can do it if needed. What do you think?

Reply:
I think the to_extensive_math!() and to_initial_amounts() converters in reactionsystem.jl are generally useful beyond MTK and therefore should remain in SBML.jl. @exaexa : do you agree? Can you rewrite them so that they no longer need mutable structs? Or shall I give it a try (e.g. putting them into utils.jl)?